### DISCUSSION TITLE
Update-MgContract unknown permission required

### DISCUSSION THREAD

**Discussion by maondrej** [microsoftgraph/msgraph-sdk-powershell#3049] [Q&A]
Dear Community, 

I'm wondering how to use the Update-MgContract cmdlet. Since there are no permissions defined, I have no idea which permissions to request. I also did not find a reference in API for update operation, just list and get. [docs.](https://learn.microsoft.com/en-us/graph/api/resources/contract?view=graph-rest-1.0)

So, usage of this cmdlet is mystery for me. Does anyone have experience with it? 

```
Find-MgGraphCommand -Command "Update-MgContract" | select Permissions

Permissions
-----------
{}
```

Calling with "Directory.ReadWrite.All" permission ends up with error: **Update-MgContract_UpdateExpanded: Insufficient privileges to complete the operation.**

Thanks a lot

**Comment by git-commit-amen** [comment]
I also have this question. Similarly, doing this via the graph API produces the same result.

`PATCH https://graph.microsoft.com/v1.0/myorganization/contracts/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`
`
{
	"displayName":"foo"
}
`

Returns:
`
{
	"@odata.context": "https://graph.microsoft.com/v1.0/$metadata#contracts/$entity",
	"id": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
	"deletedDateTime": null,
	"contractType": "ResellerPartner",
	"customerId": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
	"defaultDomainName": "foo.onmicrosoft.com",
	"displayName": "foo"
}
`

Nothing in Microsoft docs anywhere which even admits this endpoint & method combination exists. But it does.
Strange.
